### ISSUE TITLE
Bug: Argument of type 'Sprite' is not assignable to parameter of type 'DisplayObject'.

### ISSUE THREAD

**Issue by keegit** [pixijs/pixijs#10240]
### Current Behavior

The following code produces a type error type error mentioned in the title.

```typescript
import * as PIXI from 'pixi.js';

let app = new PIXI.Application<HTMLCanvasElement>({ 
        resizeTo: window,
        backgroundColor: 0x2980b9
});
document.body.appendChild(app.view); // Error: Argument of type 'Sprite'..

let sprite = PIXI.Sprite.from('breaker.png')
sprite.scale.set(5, 5)
app.stage.addChild(sprite);

let elapsed = 0.0;
app.ticker.add((delta) => {
    elapsed += delta;
    sprite.x = 100.0 + Math.cos(elapsed/25.0) * 25.0;
});
```
I am able to run `tsc` and produce an `app.js` file that I can embed in a `index.html` file and serve that - but I do not see anything when opening the browser. 

### Expected Behavior

I would expect the code to compile and run and display something in the browser.

### Steps to Reproduce

I have included this repository as an example.

1. Clone the repository
2. Open the .devcontainer using VSCode
3. Run `pnpm install`
4. Run `./mongoose -d src`
5. Open the browser at `localhost:8000`

### Environment

- **`pixi.js` version**: 7.4.0
- **Browser & Version**: Firefox 122.0.1
- **OS & Version**: Ubuntu 22.04
- **Running Example**: [example](https://github.com/keegit/PixiJS-bug-example/tree/develop)

### Possible Solution

Not sure of one. Fairly new to PixiJS and JS in general.

**Comment by bigtimebuddy** [comment]
Did you check out #9747?

**Comment by bigtimebuddy** [comment]
Thanks for digging in. I'm not sure I'm familiar with the mongoose too. This is not a devtool that has popped up for us yet in the issues. Typically our audience uses Webpack, Vite and Parcel. So it's hard for me to speak to the ReferenceError.

**Comment by keegit** [comment]
Closing - was resolved by switching to Pixi 5.3.12.

**Comment by muhajirdev** [comment]
So it didn't work with pixi 7 on your side @keegit ?
